internal/config/config.go

```diff
@@ -574,6 +574,11 @@ func addressPoolServiceAllocationsFromCR(p metallbv1beta1.IPAddressPool, namespa
 		return nil, err
 	}
 	serviceAllocations := &ServiceAllocation{Priority: p.Spec.AllocateTo.Priority, Namespaces: poolNamespaces}
+	if len(poolNamespaces) == 0 && len(p.Spec.AllocateTo.NamespaceSelectors) == 0 && len(p.Spec.AllocateTo.ServiceSelectors) == 0 {
```

Comment:
can we put this above (right after the `if p.Spec.AllocateTo == nil`), and instead `len(poolNamespaces)` use `len(p.Spec.AllocateTo.Namespaces)` ?